dali/operators/util/pad_test.cc

```diff
@@ -0,0 +1,168 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/pipeline/data/tensor.h"
+#include "dali/test/dali_operator_test.h"
+#include "dali/test/dali_operator_test_utils.h"
+
+namespace dali {
+
+namespace testing {
+
+// 1-d elements: {{5},{2},{3}} -> {{5}, {5}, {5}}
+std::vector<std::vector<float> > batch1 = {{3, 4, 2, 5, 4},
+                                           {2, 2},
+                                           {3, 199, 5}};
+std::vector<std::vector<float> > padded_batch1 = {{3, 4, 2, 5, 4},
+                                                  {2, 2, -1, -1, -1},
+                                                  {3, 199, 5, -1, -1}};
+
+// 2-d elements: {{2, 4}, {2, 2}} -> {{2, 4}, {2, 4}}
+std::vector<std::vector<int> > batch2 = {{1, 2 , 3, 4,
+                                          5, 6, 7, 8},
+                                         {1, 2,
+                                          4, 5}};
+std::vector<std::vector<int> > padded_batch2 = {{1, 2 , 3, 4,
+                                                 5, 6, 7, 8},
+                                                {1, 2, 42, 42,
+                                                 4, 5, 42, 42}};
+
+
+// 2-d elements: {{2, 4}, {3, 2}} -> {{3, 4}, {3, 4}}
+std::vector<std::vector<int> > batch3 = {{1, 2 , 3, 4,
+                                          5, 6, 7, 8},
+                                         {1, 2,
+                                          4, 5,
+                                          7, 8}};
+std::vector<std::vector<int> > padded_batch3 = {{1,  2,  3,  4,
+                                                 5,  6,  7,  8,
+                                                 0, 0, 0, 0},
+                                                {1,  2,  0, 0,
+                                                 4,  5,  0, 0,
+                                                 7,  8,  0, 0}};
+
+template <typename T>
+std::vector<std::vector<T> > GetPaddedBatchForaxes(std::vector<int> axes) {
+  return {};
+}
+
+template <>
+std::vector<std::vector<float> > GetPaddedBatchForaxes(std::vector<int> axes) {
+  if (axes[0] == 0) {
+    return padded_batch1;
+  }
+  return {};
+}
+
+template <>
+std::vector<std::vector<int> > GetPaddedBatchForaxes(std::vector<int> axes) {
+  if (axes.empty()) {
+    return padded_batch3;
+  } else if (axes[0] == 1) {
+    return padded_batch2;
+  }
+  return {};
+}
+
+class PadTest : public testing::DaliOperatorTest {
+  GraphDescr GenerateOperatorGraph() const override {
+    GraphDescr graph("Pad");
+    return graph;
+  }
+};
+
+class PadBasicTest : public PadTest {};
+class Pad2DTest : public PadTest {};
+class PadAllAxesTest : public PadTest {};
+
+
+std::vector<Arguments> basic_args = {{{"fill_value", -1.0f}, {"axes", std::vector<int>{0}}}};
+
+std::vector<Arguments> two_d_args = {{{"fill_value", 42.0f}, {"axes", std::vector<int>{1}}}};
```

Comment:
At least manually run a test which tests multiple, but not all axes.

Reply:
Test added